config/auth_u2m.go

```diff
@@ -0,0 +1,198 @@
+package config
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"os/exec"
+	"strings"
+
+	"github.com/databricks/databricks-sdk-go/credentials"
+	"github.com/databricks/databricks-sdk-go/credentials/cache"
+	"github.com/databricks/databricks-sdk-go/credentials/oauth"
+	"github.com/databricks/databricks-sdk-go/logger"
+)
+
+// U2MCredentials is a credentials strategy that uses the U2M OAuth flow to
+// authenticate with Databricks.
+//
+// To authenticate with U2M OAuth, the user must already have an existing OAuth
+// session. The specific OAuth session is indicated by the OAuth argument
+// provided by GetOAuthArg. By default, the OAuth argument is determined by the
+// account host and account ID or workspace host in the Config.
+//
+// Error handling for this strategy is controlled by the ErrorHandler field. If
+// ErrorHandler is not specified, any error will cause Configure() to return said
+// error.
+type U2MCredentials struct {
+	// Auth is the persistent auth object to use. If not specified, a new one will
+	// be created, using the default cache and locker.
+	Auth *oauth.PersistentAuth
+
+	// GetOAuthArg is a function that returns the OAuth argument to use for
+	// loading the OAuth session token. If not specified, the OAuth argument is
+	// determined by the account host and account ID or workspace host in the
+	// Config.
+	GetOAuthArg func(context.Context, *Config) (oauth.OAuthArgument, error)
+
+	// ErrorHandler controls the behavior of Configure() when loading the OAuth
+	// token fails. If not specified, any error will cause Configure() to return
+	// said error.
+	ErrorHandler func(context.Context, *Config, oauth.OAuthArgument, error) error
+
+	name string
+}
+
+// Name implements CredentialsStrategy.
+func (u U2MCredentials) Name() string {
+	if u.name != "" {
```

Comment:
This allows the unexported databricks-cli implementation to continue to use its existing name "databricks-cli", but if other users extend this strategy, the name will become `oauth-u2m`. We don't want users to choose a new name for this for consistency in our telemetry.